src/evo/cbtx.cpp

```diff
@@ -28,10 +28,11 @@ bool CheckCbTx(const CTransaction& tx, const CBlockIndex* pindexPrev, TxValidati
         return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-cbtx-invalid");
     }
 
-    CCbTx cbTx;
-    if (!GetTxPayload(tx, cbTx)) {
-        return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-cbtx-payload");
+    auto opt_cbTx = GetTxPayload<CCbTx>(tx);
+    if (!opt_cbTx) {
+        return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-cbtx-payload1");
     }
+    auto& cbTx = *opt_cbTx;
```

Comment:
```suggestion
    const auto& cbTx = *opt_cbTx;
```

Reply:
There are so many invocations like this that I don't want to spend the time making them all const.. There's no real benefit in making them const either imo

Reply:
linter was yelling about this one

Reply:
With old interface of GetTxPayload it indeed can't be const; because you *need* to change value after initialization.

But after this refactoring I vote to make all of them const; because there's not intention to change most (or maybe all) of them.